rollup.config.mjs

```diff
@@ -24,6 +37,15 @@ export default {
     }),
     typescript({
       tsconfig: 'tsconfig.build.json',
+      compilerOptions: {
+        /**
+         * Rollup wants declarations in the same directory as the output folder,
+         * but since we ship both cjs and esm in separate folders, this will throw an error.
+         * Declarations are hence built separately using tsc into lib/types
+         */
+        declaration: false,
+        declarationDir: undefined,
+      },
```

Comment:
There's already an issue raised in the plugin repo https://github.com/rollup/plugins/issues/1230#issuecomment-1324735742